static/js/lib/url.js

```diff
@@ -57,11 +57,11 @@ export const embedlyResizeImage = (key: string, url: string, height: number) =>
     url
   )}&height=${height}&grow=false&errorurl=${blankThumbnailUrl()}`
 
+export const absolutizeURL = (url: string) =>
+  new URL(url, window.location.origin).toString()
+
 export const postPermalink = (post: Post): string =>
-  new URL(
-    postDetailURL(post.channel_name, post.id),
-    window.location.origin
-  ).toString()
+  absolutizeURL(postDetailURL(post.channel_name, post.id))
```

Comment:
I noticed that the post slug isn't being included but probably should be (for example, `/c/mypost/u` vs `/c/mypost/u/myslug`).  Not the same issue the PR is addressing, but wondering if it should be fixed here anyway since the line of code is being modified?

```python
absolutizeURL(postDetailURL(post.channel_name, post.id, post.slug))
```

Reply:
yeah, I think that's a good idea, I'll go ahead and add it